### PR TITLE
rm poetry from containers

### PR DESCRIPTION
**Pull request by AndysWorth**
- Remove poetry inside gear and test container
- Add python 3.8 environment to run gear code
- Update to recent Flywheel libraries
- Fix bug when running on HPC
- Add gear-save-output-as-subfolders option
- Fix bug in saving resources used as custom information
- Remove gear-keep-ouput config option

After this branch is merged, other permanent "version" branches will be updated/created that will use the main branch for everything except the following files, which will define each different version:
- Dockerfiles
- manifest.json
- pyproject.toml
- README

See the "Contributing" section at the end of the README for a description of the state of this repository.  In short this repository is in a state of transition.  It is currently stable to allow the development of the "version" branches.  After that, a great deal of tech debt will be removed by transition to GitLab and the new best practices.  At that time, only the main branch will have to be extensively modified and the "version" branches will require minimal changes.

### CODE REVIEW

**Review by dpark6060 — CHANGES_REQUESTED** [review]
looks good, but I had some suggestions regarding confusing function names.

File: README.md [hunk 1]
```diff
@@ -166,3 +170,20 @@ as they are either handled internally or cannot be implemented.
 --fs-license-file
 --work-dir
 --clean-workdir
+
+### Contributing
+
+This repository will eventually be moved to GitHub and will use Flywheel's best practices for gears.  It is currently in an intermediate state between old development practices and the new ones.  The goal is towards this [Best Practice](https://gitlab.com/flywheel-io/scientific-solutions/gears/templates/skeleton/-/blob/main/CONTRIBUTING.md) where poetry and pre-commit are used with GitLab CI.  Previously, this gear was designed from the [bids-app-template](https://gitlab.com/flywheel-io/scientific-solutions/gears/templates/bids-app-template/-/blob/main/README.md).  The old template relied on scripts in `./tests/bin/` to build the gear and do testing inside the running gear. That is the old template and a new one will be developed soon.  No work should be done in this repository towards the new best practices until that new template has been created.
```
[the remark this answers is not in the document]
Comment (AndysWorth) [reply README.md:176]: Yup.  Good catch.

File: run.py [hunk 2]
```diff
@@ -454,11 +458,14 @@ def main(gtk_context):
 
     # make sure /flywheel/v0 is writable, use a scratch directory if not
     with flywheel_gear_toolkit.GearToolkitContext() as gtk_context:
+        gtk_context.log_config()
         scratch_dir = run_in_tmp_dir(gtk_context.config["gear-writable-dir"])
```
[the remark this answers is not in the document]
Comment (AndysWorth) [reply run.py:462]: That's a good suggestion.  It used to always run in the temp directory.  But Pablo has a new way of doing this and the method will be in the gear-toolkit so this function will go away once we get this repo onto GitLab

File: pyproject.toml [hunk 3]
```diff
@@ -1,16 +1,16 @@
 [tool.poetry]
 name = "bids-fmriprep"
-version = "1.4.0_20.2.7"
+version = "1.4.2+20.2.7"
```
[the remark this answers is not in the document]
Comment (AndysWorth) [reply pyproject.toml:3]: Yes, exactly.  I have not heard that there is a better way

**Review by dpark6060 — APPROVED** [review]

**Pull request merged**